Tools/Sources/needletools/Utilities/Version.swift

```diff
@@ -0,0 +1,140 @@
+/// `String`.
+enum VersionErrors: Error {
+    /// If the original value is not in the format of `major.minor.patch`,
+    /// where all components are numbers only.
+    case invalidFormat
+}
+
+/// The utilities of package version.
+class Version: Comparable {
+
+    /// The major part.
+    let major: Int
+    /// The minor part.
+    let minor: Int
+    /// The patch part.
+    let patch: Int
+    /// The string representation of this version.
+    lazy var stringValue = "v\(major).\(minor).\(patch)"
+
+    /// Initializer.
+    ///
+    /// - parameter string: The `String` to parse a version from.
+    /// - throws: `VersionErrors.invalidFormat` if the given `String` value
+    /// is not in the format for `major.minor.patch` where all parts are
+    /// digits.
+    init(string: String) throws {
+        let parts = string.split(separator: ".")
+            .map { (substring: Substring) -> String in
+                var string = String(substring)
+                string.removeAll { (c: Character) -> Bool in
+                    let cSet = CharacterSet(charactersIn: String(c))
+                    return CharacterSet.decimalDigits.intersection(cSet).isEmpty
+                }
+                return string
+        }
+        if parts.count != 3 {
+            throw VersionErrors.invalidFormat
+        }
+
+        major = Int(parts[0])!
+        minor = Int(parts[1])!
+        patch = Int(parts[2])!
+    }
+
+    /// Set this version as the current version.
+    ///
+    /// - throws: `VersionErrors.invalidFormat` if the version file parsing
+    /// failed.
+    func setAsCurrent() throws {
+        if let versionFileContent = Version.versionFileContent, let versionFileVersionStartIndex = Version.versionFileVersionStartIndex, let versionFileVersionEndIndex = Version.versionFileVersionEndIndex {
+            let newContent = versionFileContent.replacingCharacters(in: versionFileVersionStartIndex..<versionFileVersionEndIndex, with: "\(major).\(minor).\(patch)")
+            try newContent.write(toFile: Paths.versionFile, atomically: true, encoding: .utf8)
+        } else {
+            throw VersionErrors.invalidFormat
+        }
+    }
+
+    /// Retrieve the current version of the release.
+    ///
+    /// - returns: The current `Version`.
+    /// - throws: `VersionErrors.invalidFormat` if the version file parsing
+    /// failed.
+    static func currentVersion() throws -> Version {
+        if let versionFileContent = versionFileContent, let versionFileVersionStartIndex = versionFileVersionStartIndex, let versionFileVersionEndIndex = versionFileVersionEndIndex {
+            let stringValue = String(versionFileContent[versionFileVersionStartIndex..<versionFileVersionEndIndex])
+            return try Version(string: stringValue)
+        } else {
+            throw VersionErrors.invalidFormat
+        }
+    }
+
+    // MARK: - Comparable
+
+    static func < (lhs: Version, rhs: Version) -> Bool {
+        if lhs.major < rhs.major {
```

Comment:
nit: `return lhs.major < rhs.major || lhs.minor < rhs.minor || lhs.patch < rhs.patch` ?

Reply:
That's not enough right? If the current version is 2.2.2, and we give it a new version of 3.2.1, this would say the new version is less than the current version since the new patch '1` is less than the current patch `2`. But of course, we need to prioritize major, then minor and finally patch.

Reply:
I noticed the current implementation is also wrong. I'll fix it.